### PR TITLE
Fixing storybook build task that was preventing PR deploy sites from being bundled correctly

### PR DESCRIPTION
**Pull request by khmakoto**
#### Pull request checklist

- [ ] Addresses an existing issue: Fixes #0000
- [ ] Include a change request file using `$ yarn change`

#### Description of changes

This PR fixes the storybook build task that was preventing PR deploy sites from being bundled correctly

#### Focus areas to test

(optional)

### CODE REVIEW

File: scripts/just.config.ts [hunk 1]
```diff
@@ -44,6 +44,20 @@ function basicPreset() {
   option('push', { default: true });
 }
 
+/** Resolve whereas a storybook config + stories exist for a given path */
+function checkForStorybookExistence() {
+  const dir = process.cwd().replace(/\\/g, '/');
+  const packageName = dir.substring(dir.lastIndexOf('/') + 1);
```
Comment (ecraig12345) [inline scripts/just.config.ts:50]: You can do this with `path.basename(dir)`

File: scripts/just.config.ts [hunk 2]
```diff
@@ -44,6 +44,20 @@ function basicPreset() {
...
+
+  // Returns true if the current package has a storybook config or the examples package has a storybook config and
+  // contains a folder with the current package's name.
+  return (
+    !!resolveCwd('./.storybook/main.js') ||
+    (!!resolveCwd('../examples/.storybook/main.js') &&
+      fs.existsSync(path.join(process.cwd(), `../examples/${packageName}`)))
```
Comment (ecraig12345) [inline scripts/just.config.ts:57]: Probably better to use the path from the repo root (`findGitRoot()`) here to be safe

File: scripts/storybook/webpack.config.js [hunk 3]
```diff
@@ -74,7 +74,12 @@ module.exports = (/** @type {webpack.Configuration} */ config) => {
 
   config.resolve.extensions.push('.ts', '.tsx');
 
-  config.resolve.alias = getResolveAlias();
+  config.resolve.alias = {
+    ...getResolveAlias(),
+    '@fluentui/examples$': path.join(__dirname, '../../packages/examples/src'),
```
Comment (ecraig12345) [inline scripts/storybook/webpack.config.js:79]: Not super important but it's better to use `path.resolve` to get a path without the `../`

File: scripts/tasks/storybookTask.js [hunk 4]
```diff
@@ -41,10 +41,14 @@ module.exports.buildStorybookTask = function buildStorybookTask(options) {
...
 
+    const localConfigDir = path.join(process.cwd(), '.storybook');
+
     await storybook({
       mode: 'static',
       staticDir: [path.join(process.cwd(), 'static')],
-      configDir: path.join(process.cwd(), '.storybook'),
+      configDir: fs.existsSync(localConfigDir)
+        ? localConfigDir
+        : path.resolve(__dirname, '../../packages/examples/.storybook'),
```
Comment (ecraig12345) [inline scripts/tasks/storybookTask.js:51]: Again it would be a little cleaner to do this from `findGitRoot()`

**Review by ecraig12345 — COMMENTED** [review]
Thanks for figuring this out! A few suggestions.

**Review by dzearing — COMMENTED** [review]
Works locally, A++++++ would recommend

**Review by dzearing — APPROVED** [review]

**Review by ecraig12345 — APPROVED** [review]
Thanks!

File: scripts/tasks/storybookTask.js [hunk 5]
```diff
@@ -41,14 +42,19 @@ module.exports.buildStorybookTask = function buildStorybookTask(options) {
...
       staticDir: [path.join(process.cwd(), 'static')],
-      configDir: path.join(process.cwd(), '.storybook'),
+      configDir: fs.existsSync(localConfigDir)
+        ? localConfigDir
+        : path.join(findGitRoot(), 'packages/examples/.storybook'),
       outputDir: path.join(process.cwd(), 'dist-storybook'),
       quiet,
       port: port || 3000,
       ci,
+      debugWebpack: true,
```
Comment (ecraig12345) [inline scripts/tasks/storybookTask.js:57]: Was this supposed to be checked in?

Comment (khmakoto) [reply scripts/tasks/storybookTask.js:57]: Nope, I'm still dealing with a bunch of deploy issues and I'm using that to debug. I think it got pushed as I tried to get something working for David, but rest assured that I'll remove it by the time I merge.

**Pull request merged**